timemachine/md/minimizer.py

```diff
@@ -172,15 +183,94 @@ def minimize_host_4d(mols, host_system, host_coords, ff, box, mol_coords=None) -
     final_coords = fire_minimize(x, u_impls, box, 50)
     for impl in u_impls:
         du_dx, _ = impl.execute(final_coords, box)
-        norm = np.linalg.norm(du_dx, axis=-1)
-        if not np.all(norm < MAX_FORCE_NORM):
-            raise MinimizationError(
-                f"Minimization failed to reduce large forces below threshold: |frc| = {norm} > {MAX_FORCE_NORM}"
-            )
+        check_force_norm(du_dx)
 
     return final_coords[:num_host_atoms]
 
 
+def make_gpu_impl(potentials):
+    """return bound impl of a SummedPotential constructed from potentials"""
+
+    params = [p.params for p in potentials]
+    flat_params = np.concatenate([param.reshape(-1) for param in params])
+
+    generic_potentials = [generic.from_gpu(p) for p in potentials]
+    summed_potential = generic.SummedPotential(generic_potentials, params)
```

Comment:
nit: could this be simplified by using `timemachine.lib.potentials.SummedPotential` directly, rather than converting from GPU to generic and back to GPU?

Reply:
Ahh, good call! Will fix, no reason for the round-trip -- just this is the code snippet I had most readily at-hand.

Reply:
Round-trip avoided in https://github.com/proteneer/timemachine/pull/978/commits/11ab17763e24a875e1711e40ef3e9010a77deac4